pages/box-office/_performanceId/sell/complete.vue

```diff
@@ -34,41 +60,74 @@ export default {
       type: Booking,
     },
   },
-  async mounted() {
+  data() {
+    return {
+      checkedIn: false,
+      currentTime: new Date(),
+    }
+  },
+  mounted() {
     if (!this.booking.reference) return this.$router.push('../')
 
-    try {
-      await performMutation(
-        this.$apollo,
-        {
-          mutation: CheckInTickets,
-          variables: {
-            reference: this.booking.reference,
-            performanceId: this.booking.performance.id,
-            tickets: this.booking.tickets.map((ticket) => {
-              return {
-                ticketId: ticket.id,
-              }
-            }),
-          },
-        },
-        'checkInBooking'
-      )
-      successToast.fire({
-        timer: 4000,
-        title: 'Tickets automatically checked in',
-      })
-    } catch (e) {
-      errorToast.fire({
-        title: 'Unable to check in tickets automatically',
-      })
-    }
+    if (this.canAutoCheckIn) this.changeTicketStatus(true)
   },
   beforeDestroy() {
     // Remove stored booking ID
     this.$store.commit('box-office/SET_IN_PROGRESS_BOOKING_ID', null)
   },
   methods: {
+    canAutoCheckIn() {
+      return this.performanceDoorsDiffMinutes <= 15
+    },
+    performanceDoorsDiffMinutes() {
+      return (
+        (Date.parse(this.booking.performance.doorsOpen) - new Date()) / 60000
+      )
+    },
+    async changeTicketStatus(checkingIn) {
+      let checkInMutationGQL = CheckInTickets
+      let checkInMutationName = 'checkInBooking'
+      let sucsessMessage = this.autoCheckIn
+        ? 'Tickets automatically checked in'
+        : 'Tickets checked in'
+      let errMessage = 'Unable to check in tickets'
+
+      if (!checkingIn) {
+        checkInMutationGQL = UnCheckInTickets
+        checkInMutationName = 'uncheckInBooking'
+        sucsessMessage = 'Tickets un-checked in'
+        errMessage = 'Unable to un-check in tickets'
+      }
```

Comment:
```suggestion
      let mutationGql = checkingIn ? CheckInTickets : UnCheckInTickets
      let mutationName = checkingIn ? 'checkInBooking' : 'uncheckInBooking'
      let sucsessMessage = this.autoCheckIn
        ? 'Tickets automatically checked in'
        : checkingIn  ? 'Tickets checked in' : 'Tickets un-checked in'
      let errMessage =  checkingIn ? 'Unable to check in tickets' : 'Unable to un-check in tickets'
```

Reply:
Should also be able to move these variables in line now (i.e. where they are used)